src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -127,6 +131,45 @@ public Date read(JsonReader jsonReader) throws IOException {
     }
 
     /**
+     * TypeAdaptorFactory creating TypeAdapters for Optional
+     */
+    private class OptionalTypeAdapterFactory implements TypeAdapterFactory {
+
+        @SuppressWarnings("unchecked")
+        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> typeToken) {
+            Type type = typeToken.getType();
+            boolean isOptional = typeToken.getRawType() == Optional.class;
+            boolean isParameterized = type instanceof ParameterizedType;
+            if (isOptional && isParameterized) {
+                Type elementType = ((ParameterizedType) type).getActualTypeArguments()[0];
+                TypeAdapter<?> elementAdapter = gson.getAdapter(TypeToken.get(elementType));
+                return (TypeAdapter<T>) optionalAdapter(elementAdapter);
+            } else {
+                return null;
+            }
+        }
+
+        private <A> TypeAdapter<Optional<A>> optionalAdapter(TypeAdapter<A> innerAdapter) {
+            return new TypeAdapter<Optional<A>>() {
+                public Optional<A> read(JsonReader in) throws IOException {
+                    if (in.peek() == JsonToken.NULL) {
+                        in.nextNull();
+                        return Optional.empty();
+                    } else {
+                        A value = innerAdapter.read(in);
+                        return Optional.of(value);
+                    }
+                }
+
+                public void write(JsonWriter out, Optional<A> optional) throws IOException {
```

Comment:
Aren't `read()` and `write()` overrides from `TapeAdapter`? Let's put the `@Override` annotation?